src/main/java/tetris/player/Player.java

```diff
@@ -17,27 +24,73 @@ public int getMove(State s) {
 
     int index = -1;
 
+    ExecutorService executor = Executors.newFixedThreadPool(legalMoves.length);
+    ArrayList<Future<Double>> avgValues = new ArrayList<>();
+
     for (int i = 0; i < legalMoves.length; i++) {
       NextState nextState = new NextState(s);
-
       nextState.makeMove(i, legalMoves);
+      PlyRunner runner = new PlyRunner(nextState);
+      Future<Double> value =  executor.submit(runner);
+      avgValues.add(value);
+    }
+    executor.shutdown();
 
-      if (nextState.hasLost()) {
-        continue;
-      }
-
-      double val = heuristic.getValue(nextState);
-      if (val > maxValue) {
-        maxValue = val;
-        index = i;
+    for (int i = 0;i < avgValues.size(); i++) {
+      try{
+        Double val = avgValues.get(i).get();
+        if (val > maxValue) {
+          maxValue = val;
+          index = i;
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
       }
     }
-
     // Every move leads to a loss.
     if (index == -1) {
       index = 0; // make a random move
     }
 
     return index;
   }
+
+  public double spamPly(NextState s, int depth) {
```

Comment:
can we not name it spamPly

Reply:
this function doesn't account for the fact that we know the current nextPiece. Also I don't see depth anywhere.